### ISSUE TITLE
UI: display units for tables (big number treatment)

### ISSUE THREAD

**Issue by alistaire47** [conbench/conbench#1380]
Feedback:

> vary the result unit depending on the result value ("1772000000.0 B/s" is less readable than either "1.72 GB/s" or "1772 MB/s").

Example: https://conbench.ursa.dev/compare/runs/dd1cbc13065b4e44a998a12aca4cb108...2e63f05bf950435cbc8425fb57304a57/

<img width="451" alt="image" src="https://github.com/conbench/conbench/assets/1693477/acfeebb8-9744-4374-af11-9e2d4690478c">

We just standardized input units, but display units for tables could vary from those to make them more human-readable. We should try not to make our sorting issue worse in the process, though.

**Comment by jgehrcke** [comment]
Yes of course. Indeed, in my last iteration I was prioritizing sortability and correctness. Also: https://github.com/conbench/conbench/issues/1333

One decision we need to do: scientific notation (keeping the basic unit) or putting the order of magnitude into the unit? I think I like the former more in our context, i.e. sth like `1.7e9 B/s`.

More work on this should be enabled by https://github.com/conbench/conbench/issues/1334 where I write 

> I see this as a requirement for better number formatting: https://github.com/conbench/conbench/issues/1333

This is a bit of a philosophical stance; we need to stop putting patch onto patch onto patch in the 'augment' family of functions (if you have worked on those you know what I mean), but get rid of those first and then generate the outcome we want.

**Comment by jgehrcke** [comment]
If we choose the scientific notation path then I think we may want to use https://numpy.org/doc/stable/reference/generated/numpy.format_float_scientific.html#numpy-format-float-scientific and not `sigfig`. Reminder: sigfig is really cool for producing a string representing uncertainty (e.g., `(5.0 +/- 1.2) m/s`), but it's also slow as hell and cannot/should not be used for live-rendering hundreds of numbers (also see https://github.com/conbench/conbench/issues/1218 and https://github.com/conbench/conbench/pull/1245).

**Comment by jgehrcke** [comment]
Tiny note: natural limitation: we cannot assume a single/definite unit or order of magnitude across all values reported within one column. That is, certain brain cycles are always required when quickly browsing numbers and trying to compare them.
